packages/devtools_app/lib/src/screens/network/network_request_inspector_views.dart

```diff
@@ -201,10 +207,122 @@ class HttpViewTrailingCopyButton extends StatelessWidget {
   }
 }
 
+/// A DropDownButton for selecting [NetworkResponseType].
+///
+/// If there is no content to visualise, the drop down will not show. Drop down
+/// values will update as the request's data is updated.
+class HttpResponseTrailingDropDown extends StatefulWidget {
+  const HttpResponseTrailingDropDown(
+    this.data, {
+    super.key,
+    required this.currentresponseType,
+    required this.onNewresponseTypeSelected,
+  });
+  final ValueListenable<NetworkResponseType> currentresponseType;
+  final DartIOHttpRequestData data;
+  final ValueChanged<String> onNewresponseTypeSelected;
+
+  @override
+  State<HttpResponseTrailingDropDown> createState() =>
+      _HttpResponseTrailingDropDownState();
+}
+
+class _HttpResponseTrailingDropDownState
+    extends State<HttpResponseTrailingDropDown> {
+  List<NetworkResponseType> availableResponseTypes = List.empty(growable: true);
+  bool visible = true;
+
+  @override
+  void initState() {
+    super.initState();
+    initialise();
+  }
+
+  @override
+  void didUpdateWidget(covariant HttpResponseTrailingDropDown oldWidget) {
+    initialise();
+    super.didUpdateWidget(oldWidget);
+  }
+
+  void initialise() {
+    availableResponseTypes.clear();
+    visible = true;
+    if ((widget.data.contentType != null &&
+            widget.data.contentType!.contains('image')) ||
+        widget.data.responseBody!.isEmpty) {
+      visible = false;
+    }
+    if (visible) {
+      availableResponseTypes.add(NetworkResponseType.auto);
+      if (isJsonDecodable()) {
+        availableResponseTypes.add(NetworkResponseType.json);
+      }
+      availableResponseTypes.add(NetworkResponseType.text);
+    }
+  }
+
+  bool isJsonDecodable() {
+    try {
+      json.decode(widget.data.responseBody!);
+      return true;
+    } catch (_) {
+      return false;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder(
+      valueListenable: widget.data.requestUpdatedNotifier,
+      builder: (_, __, ___) {
+        initialise();
```

Comment:
since we are recomputing this in build, this widget does not need to be a stateful widget. The initialization code is cheap enough that it will not hinder performance to do this in build, so I would recommend that we compute variables `visible` and `availableResponseTypes` in scope here so that we can make HttpResponseTrailingDropDown a stateless widget